### ISSUE TITLE
[Feature]: Modifier sensor

### ISSUE THREAD

**Issue by disordered** [Rem0o/FanControl.Releases#1564]
Add a sensor that allows to apply a function to existing sensor. Example functions could be add, subtract, multiply or divide.

I have a CPU socket sensor that I can't get below 78C and it usually hovers around 82C. When CPU is busy, it does climb up to 90C+ and increasing system fan in that area reduces the temp. I would like to add a modifier (scale it down) on this temp, so I can put it in a group with other CPU temps and apply a graphed fan speed on the entire area. The issue right now, is that if I have a group of max temps, and I add this socket sensor into it, it constantly blows the fans at fairly high speed.

**Comment by Rem0o** [comment]
There are various ways to address this case with existing functionalities. I would argue that modifying an existing sensor values to fit other sensors is kind of a "smell" in the first place, but here are some ways on top of my head to achieve your goal.

1) Create a file sensor, name it offset, put the offset value you want in it, create a subtract custom sensor, subtract the offset from the socket sensor.  Hide all of the above except the last one. Use the last one. 

2) Use an average sensor, with that socket sensor and all other CPU sensors, and create your fan curve accordingly to that new aggregate.

[Recommended]
3) You could mix fan curves instead.   Create a fan curve dedicated to that sensor, and mix that fan curve with others.

**Comment by disordered** [comment]
I did not realise we had "Mix" curve, I can fit that sensor with it even better. I agree it is the way, so we can close this request.

Regarding the sensor, it does fluctuate by itself, but it is fairly stable. It stays in 78-82 range most of the time, but if I create load on CPU, it slowly heats up and then takes a while to cool down (significantly longer than the CPU). From what I found, glue in that area is rated up to 97C, so I want to track it.
